### DISCUSSION TITLE
Generate Custom Report

### DISCUSSION THREAD

**Discussion by abdurraheemj** [seleniumbase/SeleniumBase#1140] [Q&A] [answered]
Hi Michael,
I am using SeleniumBase for my Test Automation Project, You are being so helpful by answering all questions. 

Requirement is " I have to test multiple urls, to look for more than 10 items for each page. Have to generate one single text file, have to mention how many urls ran,  how many items are pass, how many of them are fail.for failures have to capture the url and error message in that file.Not sure how to achieve, do I have to create separate methods for each items or it is possible to achieve by using single method?  Answer with Examples would be really appreciated. I am new to Automation. God Bless You! Thanks in Advance!

**Comment by mdmintz** [comment, marked as answer]
Hello @abdurraheemj, ``self.assert_no_404_errors()`` and ``self.assert_no_js_errors()`` can help you make assertions with a single line of code. For the report, ``--html=report.html`` on the pytest command-line will generate the report for you.
